src/app/modules/form/form.service.ts

```diff
@@ -200,3 +240,39 @@ export const getFormModelByResponseMode = (
       return EncryptedFormModel
   }
 }
+
+/**
+ * Checks if a form is accessed from within intranet and sets the property accordingly
+ * @param ip The ip of the request
+ * @param publicFormView The form to check
+ * @returns ok(PublicFormView) if the form is accessed from the internet
+ * @returns err(ApplicationError) if an error occured while checking if the ip of the request is from the intranet
+ */
+export const checkIsIntranetFormAccess = (
+  ip: string,
+  form: IPopulatedForm,
+): boolean => {
+  const isIntranetIpResult = IntranetFactory.isIntranetIp(ip).andThen(
+    (isIntranetUser) => {
+      // Warn if form is being accessed from within intranet
+      // and the form has authentication set
+      if (
+        isIntranetUser &&
+        [AuthType.SP, AuthType.CP, AuthType.MyInfo].includes(form.authType)
+      ) {
+        logger.warn({
+          message:
+            'Attempting to access SingPass, CorpPass or MyInfo form from intranet',
+          meta: {
+            action: 'checkIsIntranetFormAccess',
+            formId: form._id,
+          },
+        })
+      }
+      return ok(isIntranetUser)
+    },
+  )
+
+  // This is required becausing the factory can throw missing feature error on initialization
+  return isIntranetIpResult.unwrapOr(false)
```

Comment:
would prefer not to because the only possible error that it can throw is `MissingFeatureError` and it being `false` seems like a  sensible default. if we chain, i think we're making the caller handle something additional, which i want to avoid

Reply:
you can change it and confirm - TypeScript would complain if you were changing the function interface

Reply:
oh i misunderstood. thought you wanted me to return the wrapped result instead of just adding the unwrapOr to the end .-.